modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableManager.java

```diff
@@ -2202,7 +2220,11 @@ public CompletableFuture<Void> onUpdate(WatchEvent evt) {
                 try {
                     assert evt.single();
 
-                    return handleChangePendingAssignmentEvent(evt.entryEvent().newEntry());
+                    return recoveryFuture.thenComposeAsync(v -> {
```

Comment:
Maybe it will be enough for us to wait for the `recoveryFuture` once during recovery and not use it again?

Reply:
How will you do that? Versioned Values get completed after all components have started. I've modified the Watch code to be a little bit more optimal

Reply:
Yes, at the same time it allows you to avoid races when updating the metastore revision, which can ruin someone’s life)

Reply:
Please check the current approach, is it ok? 

> Yes, at the same time it allows you to avoid races when updating the metastore revision

What do you mean?